rclc_parameter/src/rclc_parameter/parameter_server.c

```diff
@@ -44,24 +56,43 @@ rclc_parameter_server_describe_service_callback(
   DescribeParameters_Request * request = (DescribeParameters_Request *) req;
   DescribeParameters_Response * response = (DescribeParameters_Response *) res;
 
-  size_t size = (request->names.size > param_server->parameter_list.size) ?
-    param_server->parameter_list.size :
-    request->names.size;
-
-  response->descriptors.size = size;
+  if (request->names.size > response->descriptors.capacity) {
+    response->descriptors.size = 0;
+    return;
+  }
 
-  for (size_t i = 0; i < size; i++) {
-    rclc_parameter_set_string(
-      &response->descriptors.data[i].name,
-      request->names.data[i].data);
+  response->descriptors.size = request->names.size;
 
-    Parameter * parameter = rclc_parameter_search(
+  for (size_t i = 0; i < request->names.size; i++) {
```

Comment:
```suggestion
  for (size_t i = 0; i < request->names.size; ++i) {
```
Minor detail: Google C++ code style, which is basis of ROS 2 C++ code style, says _'Use prefix increment/decrement, unless the code explicitly needs the result of the postfix increment/decrement expression.'_

Reply:
Updated: 50c4a127f19375e24892fd611547dd5ba5195e9a